src/librustc/hir/check_attr.rs

```diff
@@ -154,7 +154,22 @@ impl<'a, 'tcx> CheckAttrVisitor<'a, 'tcx> {
         let hints: Vec<_> = item.attrs
             .iter()
             .filter(|attr| attr.name() == "repr")
-            .filter_map(|attr| attr.meta_item_list())
+            .filter_map(|attr| {
+                let list = attr.meta_item_list();
+                let mut has_hints = false;
+                if let Some(ref list) = list {
+                    has_hints = !list.is_empty();
+                }
+                if !has_hints {
```

Comment:
Nit: I think this is better written as `let is_empty = list.map(|list| list.is_empty()).unwrap_or(true);` and then `if empty { ... }`.

Reply:
I caught that while reworking this. Changed it.